### PR TITLE
feat(Less): Replace mixins with variables for media queries

### PR DESCRIPTION
**Pull request by seangrant**
Updated some styles to fix stylelint problems that appear to have been ignored while using the mixins

### CODE REVIEW

File: react/Header/UserAccount/UserAccount.less [hunk 1]
```diff
@@ -127,22 +129,22 @@
...
 }
 
 .chevronSvg {
   width: 14px;
   height: 14px;
   // Begin bold icon hacks
-  .smallDeviceOnly({
-    path {
+  @media @smallDeviceOnly {
+    path { // stylelint-disable-line selector-max-type
```
Comment (markdalgleish) [inline react/Header/UserAccount/UserAccount.less]: You can avoid the comment with `:global(path)`, that's the intention of this rule in our codebase.

File: react/Header/UserAccountMenu/UserAccountMenu.less [hunk 2]
```diff
@@ -84,39 +84,39 @@
...
 
 .iconSvg {
   width: 15px;
   height: 15px;
   // Begin bold icon hacks
-  .smallDeviceOnly({
+  @media @smallDeviceOnly {
     overflow: visible;
-    path {
+    path { // stylelint-disable-line selector-max-type
```
Comment (markdalgleish) [inline react/Header/UserAccountMenu/UserAccountMenu.less]: `:global(path)` again.

File: react/Header/UserAccount/UserAccount.less (hunk 1 above)
Comment (seangrant) [inline react/Header/UserAccount/UserAccount.less]: 👍

File: react/Header/UserAccountMenu/UserAccountMenu.less (hunk 2 above)
Comment (seangrant) [inline react/Header/UserAccountMenu/UserAccountMenu.less]: 👍

**Comment by markdalgleish** [comment]
@seangrant just a heads up that I wordsmithed the PR title 😄

File: theme/layout/breakpoints.less [hunk 3]
```diff
@@ -1,14 +1,6 @@
 // Breakpoint
 @responsive-breakpoint: 740px;
 
-.smallDeviceOnly(@rules) {
-  @media only screen and (max-width: (@responsive-breakpoint - 1)) {
-    @rules();
-  }
-}
+@smallDeviceOnly: %(~"only screen and (max-width: %s)", @responsive-breakpoint - 1);
```
Comment (markdalgleish) [inline theme/layout/breakpoints.less]: While we're at it, can we rename this to `@mobile`?

File: theme/layout/breakpoints.less [hunk 4]
```diff
@@ -1,14 +1,6 @@
...
 
-.aboveSmallDevice(@rules) {
-  @media only screen and (min-width: @responsive-breakpoint) {
-    @rules();
-  }
-}
+@aboveSmallDevice: %(~"only screen and (min-width: %s)", @responsive-breakpoint);
```
Comment (markdalgleish) [inline theme/layout/breakpoints.less]: Similarly, can we rename this to `@desktop`?

**Comment by markdalgleish** [comment]
@seangrant I wonder if we should leave the old mixins for now and maintain backwards compatibility?

File: theme/layout/breakpoints.less [hunk 5]
```diff
@@ -1,6 +1,10 @@
 // Breakpoint
 @responsive-breakpoint: 740px;
 
+@mobile: %(~"only screen and (max-width: %s)", (@responsive-breakpoint - 1));
+
+@aboveSmallDevice: %(~"only screen and (min-width: %s)", @responsive-breakpoint);
```
Comment (markdalgleish) [inline theme/layout/breakpoints.less]: Can we rename this one to `@desktop`?

File: docs/src/components/Demo/Demo.less [hunk 6]
```diff
@@ -22,27 +22,25 @@
 
 .options {
   padding: @grid-row-height @grid-gutter-width (@grid-row-height * 2) @grid-gutter-width;
-
-  .aboveSmallDevice({
+  @media @desktop {
```
Comment (markdalgleish) [inline docs/src/components/Demo/Demo.less]: Nice one—now this reads the way we *talk* about this stuff, both devs and designers, which is definitely what we're going for 💯 😎

File: theme/layout/breakpoints.less [hunk 7]
```diff
@@ -1,6 +1,10 @@
 // Breakpoint
 @responsive-breakpoint: 740px;
 
+@mobile: %(~"only screen and (max-width: %s)", (@responsive-breakpoint - 1));
+
+@desktop: %(~"only screen and (min-width: %s)", @responsive-breakpoint);
+
 .smallDeviceOnly(@rules) {
```
Comment (markdalgleish) [inline theme/layout/breakpoints.less]: Sorry to nitpick this, but could you add a comment that the mixins are for backwards compatibility?

File: docs/src/components/Demo/Demo.less (hunk 6 above)
Comment (seangrant) [inline docs/src/components/Demo/Demo.less]: 👍  Added separate comment for each variable

File: theme/layout/breakpoints.less [hunk 8]
```diff
@@ -1,12 +1,18 @@
 // Breakpoint
 @responsive-breakpoint: 740px;
 
+@mobile: %(~"only screen and (max-width: %s)", (@responsive-breakpoint - 1));
+
+@desktop: %(~"only screen and (min-width: %s)", @responsive-breakpoint);
+
+// DEPRECATED: Use @mobile variable instead.
 .smallDeviceOnly(@rules) {
   @media only screen and (max-width: (@responsive-breakpoint - 1)) {
```
Comment (markdalgleish) [inline theme/layout/breakpoints.less]: Actually, just realised we should probably refactor these mixins to use the new variables too.

Comment (seangrant) [inline theme/layout/breakpoints.less]: When you say it like that, it sounds obvious, will do that now

**Pull request merged**